src/mbgl/util/i18n.hpp

```diff
@@ -6,9 +6,6 @@ namespace mbgl {
 namespace util {
 namespace i18n {
 
-/** Returns whether a character is a visible character. */
-bool isVisible(uint16_t chr);
```

Comment:
This was only used to remove trailing characters from a line, and only newline/space/zero-width space. I replaced it with trim because I think that's the semantics we're going for -- we don't want whitespace trailing/leading whitespace to affect layout. Zero-width space will get "rendered" now, although it won't affect layout unless the font we're using assigns it non-zero width.

"Invisible" sounds to me like a font property (if there's no glyph, the code point is effectively invisible, also the font may have a glyph that doesn't draw any pixels). As you note below Unicode also has lots of extra "whitespace" characters. This change doesn't affect our handling of them.